sdk/storage/Azure.Storage.Files.DataLake/src/DataLakeFileSystemClient.cs

```diff
@@ -270,7 +270,21 @@ internal DataLakeFileSystemClient(Uri fileSystemUri, HttpPipeline pipeline, Clie
             _dfsUri = GetDfsUri(fileSystemUri);
             _pipeline = pipeline;
             _clientDiagnostics = clientDiagnostics;
-            _containerClient = new BlobContainerClient(_blobUri, pipeline, clientDiagnostics, null);
+            _containerClient = BlobContainerClientHelper.Create(_blobUri, pipeline, _clientDiagnostics);
+        }
+
+        private class BlobContainerClientHelper : BlobContainerClient
```

Comment:
Nit - consider renaming to `BlobContainerClientInternals` if we're going that direction.  Ditto for the others.